moto/dynamodb/models/table_export.py

```diff
@@ -0,0 +1,111 @@
+import json
+from threading import Thread
+from typing import TYPE_CHECKING, Any, Dict, Optional
+from uuid import uuid4
+
+from moto.core.utils import gzip_compress
+from moto.utilities.utils import get_partition
+
+if TYPE_CHECKING:
+    from moto.dynamodb.models import Table
+    from moto.s3.models import S3Backend
+
+
+class TableExport(Thread):
+    def __init__(
+        self,
+        s3_bucket: str,
+        s3_prefix: str,
+        region_name: str,
+        account_id: str,
+        table_arn: str,
+        export_format: str = "DYNAMO_JSON",
+        export_type: str = "FULL_EXPORT",
+    ):
+        super().__init__()
+        self.partition = get_partition(region_name)
+        self.table_arn = table_arn
+        self.arn = f"arn:{self.partition}:dynamodb:{region_name}:{account_id}:table/{table_arn}/import/{str(uuid4()).replace('-', '')}"
+        self.s3_bucket = s3_bucket
+        self.s3_prefix = s3_prefix
+        self.status = "IN_PROGRESS"
+        self.export_format = export_format
+        self.export_type = export_type
```

Comment:
I haven't built anything for incremental export, is what is exported important here? I'm not even sure how we would go about supporting incremental exports as we'd need to be taking snapshots of the table and everything's stored in memory so I've just left it to perform the same as the full export.